### DISCUSSION TITLE
How to use IJetStreamPullAsyncSubscription, Handler is not invoked by NATS on publish

### DISCUSSION THREAD

**Discussion by rdkumavat1** [nats-io/nats.net#573] [General]
I am trying to use pull consumer with Handler, I am expecting NATS should call that handler on new message publish in the subject, but it is not happening in my case, Can you please share document or example for the same?

I have this sample code.

var consumerConfigModel = Config.ProcessConfigurationObject<ConsumerConfigModel>(consumerConfigObject);

            var configBuilder = ConsumerConfiguration.Builder();

            configBuilder
                                 .WithDurable(consumerName)
                                 .WithDeliverPolicy(DeliverPolicy.All)
                                 .WithAckPolicy(AckPolicy.Explicit)
                                 .WithReplayPolicy(ReplayPolicy.Instant)
                                 .WithAckWait(consumerConfigModel.max_ack_wait_duration)
                                 .WithMaxDeliver(consumerConfigModel.max_deliver)
                                 .WithMaxAckPending(consumerConfigModel.max_pending_ack)
                                 .WithFilterSubject(subject);
                                //.WithMaxBatch(100);

            // .WithRateLimitBps(subscribers.rateLimitsBps);

            PullSubscribeOptions pullOptions = PullSubscribeOptions.Builder()
                    .WithConfiguration(configBuilder.Build())
                    .WithStream(streamName)
                    
                    .Build();

            var subscriptionInfo = jetStream.PullSubscribeAsync(subject, handler, pullOptions);

**Comment by neilalexander** [comment]
This question is probably better asked in [nats-io/nats.net](https://github.com/nats-io/nats.net) but seeing as I don't seem to be able to transfer the discussion there, paging @mtmk instead.

**Reply by rdkumavat1** [reply]
Thank you, I have reposted the discussion on https://github.com/nats-io/nats.net

**Comment by mtmk** [comment]
cc @scottf

**Comment by scottf** [comment]
@rdkumavat1 @neilalexander @mtmk This is a V1 question, I moved the original here: https://github.com/nats-io/nats.net.v1/discussions/914 , this was re-post

**Reply by mtmk** [reply]
thanks @scottf 🙏
